### PR TITLE
[None][fix] Reliability fixes for MTP with DSA and support host cache offload for DSA

### PR DESCRIPTION
**Pull request by dmtri35**
## Summary by CodeRabbit

* **New Features**
  * Added `indexer_rope_interleave` configuration option for controlling rotational embedding interleaving behavior in attention operations.

* **Bug Fixes**
  * Improved safety checks for block table access to prevent out-of-bounds errors.
  * Added kernel synchronization to prevent potential deadlocks in concurrent stream execution.

* **Improvements**
  * Enhanced memory pool management for sparse attention operations.
  * Refined padding and alignment safeguards for more robust block table handling.

## Description

Important fixes that we landed internally on the way to productionize [GLM5](https://www.baseten.co/blog/how-we-built-the-fastest-glm-5-api/):
- Support indexer host cache for DSA.
- Fix a fairly often hang at `fp8_mqa_logits` due to it being a persistent kernel, so we added a sync before launching it.
- Fix a quality issue that comes up when running MTP + overlap scheduler + with DSA it generate garbled outputs after a while.
- Support indexer_rope_interleave (hard requirement for GLM5 or it generates garbage after a while)

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/attention_backend/sparse/dsa.py [hunk 1]
```diff
@@ -768,29 +790,66 @@ def prepare(self):
...
+            req_indices = self.req_idx_per_token[:self.num_tokens].to(
+                dtype=torch.int64)
+            seq_starts = torch.cumsum(
+                seq_lens, dim=0, dtype=torch.int64) - seq_lens.to(torch.int64)
+            token_offsets = torch.arange(
+                self.num_tokens, device=seq_lens.device,
+                dtype=torch.int64) - seq_starts[req_indices]
+
+            global_positions = start_positions[req_indices] + token_offsets
+            block_indices_in_seq = global_positions // tokens_per_block
```
Comment (yuxianq) [inline tensorrt_llm/_torch/attention_backend/sparse/dsa.py]: Should we also add a `assert (block_indices_in_seq < max_blocks).all()` check like https://github.com/NVIDIA/TensorRT-LLM/blob/994c3cd30d2bbed7db1e9722eafe28e41adae473/tensorrt_llm/_torch/attention_backend/sparse/dsa.py#L1220

File: tensorrt_llm/_torch/attention_backend/utils.py [hunk 2]
```diff
@@ -97,4 +98,5 @@ def create_attention(
         sparse_attention_config=sparse_attention_config,
         dtype=dtype,
         aux_stream=aux_stream,
+        indexer_rope_interleave=indexer_rope_interleave,
```
Comment (chang-l) [inline tensorrt_llm/_torch/attention_backend/utils.py]: Can we move this arg into `DeepSeekSparseAttentionConfig`, like other indexer configs?
https://github.com/NVIDIA/TensorRT-LLM/blob/3139ffa79826bc6c3d976cef8833f06068c20296/tensorrt_llm/llmapi/llm_args.py#L261

**Review by chang-l — COMMENTED** [review]
Should we add a test to validate DSA host-offloading for both pools (MLA KV cache + indexer-K cache)?

File: tensorrt_llm/_torch/attention_backend/utils.py (hunk 2 above)
Comment (dmtri35) [reply tensorrt_llm/_torch/attention_backend/utils.py]: yes

**Review by yuxianq — APPROVED** [review]

**Review by chang-l — APPROVED** [review]

**Review by lfr-0531 — APPROVED** [review]

**Review by StanleySun639 — APPROVED** [review]

**Review by nv-guomingz — APPROVED** [review]
LGTM for llm api part.
